packages/app-store/stripepayment/api/subscription.ts

```diff
@@ -0,0 +1,161 @@
+import { UserPlan } from "@prisma/client";
+import type { NextApiRequest, NextApiResponse } from "next";
+import Stripe from "stripe";
+
+import { checkPremiumUsername } from "@calcom/ee/lib/core/checkPremiumUsername";
+import hasKeyInMetadata from "@calcom/lib/hasKeyInMetadata";
+import prisma from "@calcom/prisma";
+import { Prisma } from "@calcom/prisma/client";
+import {
+  PREMIUM_PLAN_PRICE,
+  PREMIUM_PLAN_PRODUCT_ID,
+  PRO_PLAN_PRICE,
+  PRO_PLAN_PRODUCT_ID,
+} from "@calcom/stripe/constants";
+import { getStripeCustomerIdFromUserId } from "@calcom/stripe/customer";
+import stripe from "@calcom/stripe/server";
+
+enum UsernameChangeStatusEnum {
+  NORMAL = "NORMAL",
+  UPGRADE = "UPGRADE",
+  DOWNGRADE = "DOWNGRADE",
+}
+
+const obtainNewConditionAction = ({
+  userCurrentPlan,
+  isNewUsernamePremium,
+}: {
+  userCurrentPlan: UserPlan;
+  isNewUsernamePremium: boolean;
+}) => {
+  const isUserPremium = userCurrentPlan === UserPlan.PRO;
+  let resultCondition = UsernameChangeStatusEnum.NORMAL;
+  if (isUserPremium && isNewUsernamePremium) {
+    resultCondition = UsernameChangeStatusEnum.UPGRADE;
+  } else if (isUserPremium && !isNewUsernamePremium) {
+    resultCondition = UsernameChangeStatusEnum.DOWNGRADE;
+  }
+  return resultCondition;
```

Comment:
[Prefer early returns](https://app.gitbook.com/o/6snd8PyPYMhg0wUw6CeQ/s/VXRprBTuMlihk37NQgUU/engineering/best-practices/prefer-early-returns)

```suggestion
  if (userCurrentPlan === UserPlan.PRO) {
    if (isNewUsernamePremium) return UsernameChangeStatusEnum.UPGRADE;
    return UsernameChangeStatusEnum.DOWNGRADE;
  }
  return UsernameChangeStatusEnum.NORMAL;
```